.github/workflows/cont_integration.yml

```diff
@@ -1,23 +1,21 @@
-on: [pull_request]
+on:
+  push:
+    branches:
+      - master
+  pull_request: {}
 
 name: CI
 
 jobs:
 
-  cosmetics:
-    runs-on: ubuntu-20.04
+  build:
+    strategy:
+      matrix:
+        os: ["macos-12", "ubuntu-22.04"]
+    runs-on: ${{ matrix.os }}
     steps:
       - name: Checkout
-        uses: actions/checkout@v2
-      - name: Setup Cache
-        uses: Swatinem/rust-cache@v1.2.0
-      - name: Setup rust toolchain
-        uses: actions-rs/toolchain@v1
-        with:
-          toolchain: stable
-          override: true
-          profile: minimal
-          components: rustfmt
-      - name: Format
-        run: cargo fmt -- --check
+        uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@stable
+      - run: cargo build
```

Comment:
needs deps

```suggestion
      - uses: dtolnay/rust-toolchain@stable
      - run: sudo apt-get update && sudo apt-get install -y libudev-dev
      - run: cargo build --all-features --all-targets
```

Reply:
`apt` would be a problem on mac?

I think here is probably better to have `cargo build --no-default-features` and be sure the serial requirement is correctly feature gated

Reply:
avoiding to pull serialport and thus libudev seems harder than expected, in the meantime I went with 7b0d81515ac9328165c01eaba318d9d4a5beed32